R/bayesianmetaanalysisbinomial.R

```diff
@@ -0,0 +1,457 @@
+  }
+
+  return()
+
+}
+.bibmaCleanModel          <- function(jaspResults) {
+
+  if (!is.null(jaspResults[["model"]])) {
+    jaspResults[["model"]] <- NULL
+  }
+
+  return()
+}
+.bibmaGetData             <- function(options, dataset) {
+
+  if (!is.null(dataset)) {
+    return(dataset)
+  } else {
+    varNames <- c(options[["successesGroup1"]], options[["successesGroup2"]], options[["observationsGroup1"]], unlist(options[["observationsGroup2"]]))
+    varNames <- varNames[varNames != ""]
+
+    dataset <- readDataSetToEnd(
+      columns.as.numeric = varNames,
+      columns = if (options[["studyLabel"]] != "") options[["studyLabel"]]
+    )
+
+    if (options[["studyLabel"]] != "") {
+      dataset[[options[["studyLabel"]]]] <- as.character(dataset[[options[["studyLabel"]]]])
+      if (any(!validUTF8(dataset[[options[["studyLabel"]]]])))
+        .quitAnalysis(gettext("The study labels contain invalid characters. Please, remove them before running the analysis."))
+    }
+
+  }
+
+  return(dataset)
+}
+.bibmaGetPriors           <- function(jaspResults, options) {
+
+  if (!is.null(jaspResults[["priors"]])) {
+    return()
+  } else {
+    priors <- createJaspState()
+    priors$dependOn(.bibmaDependencies)
+    jaspResults[["priors"]] <- priors
+  }
+
+  object <- list()
+
+  for (parameter in c("effect", "heterogeneity", "baseline")) {
+    for (type in c("", "Null")) {
+
+      optionName <- switch(
+        paste0(parameter, type),
+        "effect"            = "modelsEffect",
+        "effectNull"        = "modelsEffectNull",
+        "heterogeneity"     = "modelsHeterogeneity",
+        "heterogeneityNull" = "modelsHeterogeneityNull",
+        "baseline"          = "modelsBaseline",
+        "baselineNull"      = "modelsBaselineNull"
+      )
+
+      tmp <- NULL
+      for (j in seq_along(options[[optionName]])) {
+        tmpPrior <- try(.robmaExtractPriorsFromOptions(options[[optionName]][[j]], parameter = parameter))
+        if (jaspBase::isTryError(tmpPrior))
+          .quitAnalysis(tmpPrior)
+        else
+          tmp <- c(tmp, list(tmpPrior))
+      }
+      object[[paste0(parameter, type)]] <- tmp
+
+    }
+  }
+
+  priors[["object"]] <- object
+
+  return()
+}
+# main functions
+.bibmaPriorsPlots              <- function(jaspResults, options) {
+
+  # create / access the container
+  if (!is.null(jaspResults[["priorPlots"]]))
+    priorPlots <- jaspResults[["priorPlots"]]
+  else {
+    priorPlots <- createJaspContainer(title = gettext("Prior Plots"))
+    priorPlots$dependOn("priorDistributionPlot")
+    priorPlots$position <- 2
+    jaspResults[["priorPlots"]] <- priorPlots
+  }
+
+  # extract the priors
+  if (is.null(jaspResults[["model"]]))
+    priors <- jaspResults[["priors"]][["object"]]
+  else
+    priors <- jaspResults[["model"]][["object"]][["priors"]]
+
+  # create container for each of the parameters
+  for (parameter in c("effect", "heterogeneity")) {
+
+    if (!is.null(priorPlots[[parameter]]))
```

Comment:
I would prefer that if you work with if else and one of them needs brackets that the other also gets brackets.